### ISSUE TITLE
Ignore needless_doctest_main in README?

### ISSUE THREAD

**Issue by Heliozoa** [rust-lang/rust-clippy#14369]
### Description

In one of my projects (https://github.com/Heliozoa/elm_rs) I have a Rust code block in the readme that shows a complete example program. This triggers clippy's needless_doctest_main which has the rationale of "The test can be shorter (and likely more readable) if the fn main() is left implicit."

```
❯ cargo clippy
   Compiling elm_rs_derive v0.2.3 (/home/heliozoa/Dev/elm_rs/elm_rs_derive)
    Checking elm_rs v0.2.3 (/home/heliozoa/Dev/elm_rs/elm_rs)
warning: needless `fn main` in doctest
  --> elm_rs/src/../README.md:11:1
   |
11 | / use elm_rs::{Elm, ElmEncode, ElmDecode, ElmQuery, ElmQueryField};
12 | |
13 | | #[derive(Elm, ElmEncode, ElmDecode)]
14 | | enum Filetype {
...  |
54 | |     println!("{}", output);
55 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_doctest_main
   = note: `#[warn(clippy::needless_doctest_main)]` on by default

warning: `elm_rs` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.98s
```

I think for a readme it's good to be more verbose and explicit, as well as easily copied and ran by anyone reading it, as opposed to a doctest associated with a single function. I'm also not sure if it's even possible to use `#[allow]` on just the readme file, possibly leaving allowing it for the entire library as the only option to get rid of the warning. What would you think about not checking for this lint in readme files?

### Version

```text
rustc 1.85.0 (4d91de4e4 2025-02-17)
binary: rustc
commit-hash: 4d91de4e48198da2e33413efdcd9cd2cc0c46688
commit-date: 2025-02-17
host: x86_64-unknown-linux-gnu
release: 1.85.0
LLVM version: 19.1.7
```

**Comment by samueltardieu** [comment]
Note that clippy doesn't lint the `README.md` file itself: here, the lint triggers because the `README.md` file is included by `#![doc = include_str!("../README.md")]` in `elm_rs/src/lib.rs`.

One way to solve this would be to include this file conditionally when generating the crate documentation:

```rust
#![cfg_attr(doc, doc = include_str!("../README.md"))]
```

Would that be satisfactory to you?

**Comment by Heliozoa** [comment]
Oh, that makes sense. It's been a while since I added that so I forgot about my use of `include_str`... The `cfg_attr` solution works very well! I was worried that it may cause the example code to not be used as a doctest, but of course doctests are built with the doc flag so there's no problem. Thank you for your help.
